api/testdata/complete-scenarios/test10.json

```diff
@@ -0,0 +1,5748 @@
+          "type": "textarea",
+          "props": {
+            "value": ""
+          }
+        }
+      }
+    },
+    "Hospitalizations": {
+      "type": "psychological.hospitalizations",
+      "props": {
+        "Hospitalized": {
+          "type": "branch",
+          "props": {
+            "value": "No"
+          }
+        },
+        "List": {
+          "type": "collection",
+          "props": {
+            "branch": {
+              "type": ""
+            },
+            "items": []
+          }
+        }
+      }
+    }
+  },
+  "Relationships": {
+    "Cohabitants": {
+      "type": "relationships.status.cohabitant",
+      "props": {
+        "HasCohabitant": {
+          "type": "branch",
+          "props": {
+            "value": "Yes"
+          }
+        },
+        "CohabitantList": {
+          "type": "collection",
+          "props": {
+            "branch": {
+              "type": "branch",
+              "props": {
+                "value": "No"
+              }
+            },
+            "items": [
+              {
+                "Item": {
+                  "BirthPlace": {
+                    "type": "location",
+                    "props": {
+                      "layout": "Birthplace without County",
+                      "city": "Victoria",
+                      "country": "Canada"
+                    }
+                  },
+                  "Birthdate": {
+                    "type": "datecontrol",
+                    "props": {
+                      "month": "01",
+                      "day": "01",
+                      "year": "1980",
+                      "estimated": false
+                    }
+                  },
+                  "Citizenship": {
+                    "type": "country",
+                    "props": {
+                      "value": [
+                        "Canada",
+                        "United States"
+                      ]
+                    }
+                  },
+                  "CohabitationBegan": {
+                    "type": "datecontrol",
+                    "props": {
+                      "month": "01",
+                      "day": "03",
+                      "year": "2014",
+                      "estimated": false
+                    }
+                  },
+                  "ForeignBornDocument": {
+                    "type": "foreignborndocument",
+                    "props": {
+                      "DocumentType": {
+                        "type": "radio",
+                        "props": {
+                          "value": "CertificateOfNaturalization",
+                          "checked": true
+                        }
+                      },
+                      "OtherExplanation": {
+                        "type": "textarea",
+                        "props": {
+                          "value": ""
+                        }
+                      },
+                      "DocumentNumber"
```

Comment:
I think the purpose of the additional test is to have a current spouse entry where email is not applicable, if I am following correctly. So as-is, this case doesn't appear to test anything different than other complete scenarios that have a spouse.

Reply:
I just took a glance at `spouse-present-marriage.xml`, line 42 – the XML template logic does not support the new "Not Applicable" flag in the UI. 

It needs either:
* some direct template logic like in `history-residence.xml` , line 19 **OR**
* a helper function like `monthYearOptional` (used in conjunction with conditional template logic `date-month-year-optional.xml`).

A helper function could be used in other places to make the code more defensive, e.g., `personal-references.xml`, line 12.

If you would feel more comfortable if I took on this part of the PR, let me know and we can split this PR up.

Reply:
I've made the updates as discussed